### PR TITLE
late awaken traitors get voices in their head for flavor

### PR DESCRIPTION
**Pull request by ToasterBiome**
# Document the changes in your pull request

adds an "awaken" system to traitors.

They awaken 2 minutes earlier but go through a 2 minute "awaken" sequence where it plays a spooky sound and you get schizophrenia. 

You start with an "I don't feel good" message and an ominous sound.

https://github.com/yogstation13/Yogstation/assets/5091394/f70a4334-3f17-4751-8bdf-efff10b3a485

A little bit later you get an antagonist specific "hint" and another sound. 

The hint for "Traitor":
_Remembering a tune, you slowly find the melody. Coded phrases and dark rooms flutter behind your eyelids. What could it mean? You should probably keep this to yourself._

https://github.com/yogstation13/Yogstation/assets/5091394/e62cb223-79cc-47ca-83fe-31adec676497

Finally adds a little buildup to the traitor alert courtesy of @ynot01 

This is the beginning of what might be a flavorful antag awaken system, with the hint being "different" on what antag you will eventually turn into.

Doesn't change any aspect of gameplay except giving you a hint to _maybe_ get to a safe area beforehand.

Adds more "i'm a new player and WHAT THE FUCK IS HAPPENING"

# Changelog

:cl:  ToasterBiome, Ynot01
rscadd: Adds spooky Traitor awaken system
/:cl:

### CODE REVIEW

**Comment by TerraTheUnwise** [comment]
This is cool, I like this.

**Comment by MajManatee** [comment]
concerned with people seeing the inital messages, and doing antagy things before they are fully given the role. could be a rule issue.

**Comment by ToasterBiome** [comment]
> concerned with people seeing the inital messages, and doing antagy things before they are fully given the role. could be a rule issue.

Not the antag yet, rules are pretty clear.

**Comment by Xoxeyos** [comment]
It _looks_ like it's in the code, but I want to make certain; Would admins be able to tell who would be a traitor before they awaken? 
This change does bring about the plausible scenario in which a "sleeping" traitor would engage in potentially rule-breaking activity (IC or Server Wise), wouldn't it be smart to have (and what I appear to already see) a cached list of sleeping traitors ready for admins?

**Comment by ToasterBiome** [comment]
> It _looks_ like it's in the code, but I want to make certain; Would admins be able to tell who would be a traitor before they awaken? 
> 
> This change does bring about the plausible scenario in which a "sleeping" traitor would engage in potentially rule-breaking activity (IC or Server Wise), wouldn't it be smart to have (and what I appear to already see) a cached list of sleeping traitors ready for admins?

They already get the special role tag in their mind, so you can see it in the TP before they actually awaken.

File: code/modules/antagonists/_common/antag_datum.dm [hunk 1]
```diff
@@ -20,6 +20,7 @@ GLOBAL_LIST_EMPTY(antagonists)
 	var/hud_icon = 'yogstation/icons/mob/antag_hud.dmi'
 	///Name of the antag hud we provide to this mob.
 	var/antag_hud_name
+	var/awake_stage = ANTAG_AWAKE //What stage we are of "waking up"
```
Comment (JohnFulpWillard) [inline code/modules/antagonists/_common/antag_datum.dm:23]: Shouldn't this be on the Traitor datum if that's who we want to be able to wake up?

Comment (ToasterBiome) [reply code/modules/antagonists/_common/antag_datum.dm:23]: Leaving on the generic antag datum for other antag awakenings.

**Review by Moltijoe — APPROVED** [review]

**Review by JamieD1 — APPROVED** [review]
Looks good

**Review by ynot01 — APPROVED** [review]

**Comment by ynot01** [comment]
I don't really like knowing I'm traitor before I'm traitor but whatev

**Pull request merged**